lib/context_adapter_server.js

```diff
@@ -0,0 +1,427 @@
+/*
+ * Copyright 2015 Telefónica Investigación y Desarrollo, S.A.U
+ *
+ * This file is part of context-adapter
+ *
+ * context-adapter is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * context-adapter is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with context-adapter.
+ * If not, seehttp://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License
+ * please contact with::[german.torodelvalle@telefonica.com]
+ */
+
+(function() {
+  'use strict';
+
+  var caConfig = require('./context_adapter_configuration');
+  var caLogger = require('./context_adapter_logger');
+  var caHelper = require('./context_adapter_helper.js');
+  var hapi = require('hapi');
+  var boom = require('boom');
+
+  var server;
+
+  var attendedRequests = 0;
+
+  /**
+   * Checks that every request sent to the Context Adapter includes the required headers
+   * @param {object} value Headers object
+   * @param {object} options Hapi server header validation configuration object
+   * @param {function} next Hapi server header validation continuation function
+   */
+  function validateHeaders(value, options, next) {
+    var error, message;
+
+    var meta = {
+      unicaCorrelator: value[caConfig.UNICA_CORRELATOR_HEADER] || caHelper.getUnicaCorrelator(),
+      transactionId: caHelper.getTransactionId(),
+      operationType: caHelper.getOperationType()
+    };
+
+    attendedRequests++;
+
+    if (!value['fiware-service']) {
+      message = 'error=child "fiware-service" fails because [fiware-service is required]';
+      caLogger.warn(
+        message,
+        meta
+      );
+      error = boom.badRequest(message);
+      error.output.payload.validation = {source: 'headers', keys: ['fiware-service']};
+      next(error);
+    } else if (!value['fiware-servicepath']) {
+      message = 'child "fiware-servicepath" fails because [fiware-servicepath is required]';
+      caLogger.warn(
+        message,
+        meta
+      );
+      error = boom.badRequest(message);
+      error.output.payload.validation = {source: 'headers', keys: ['fiware-servicepath']};
+      next(error);
+    }
+    next();
+  }
+
+  /**
+   * Starts the server asynchronously
+   * @param {string} host The Context Adapter server host
+   * @param {string} port The Context Adapter server port
+   * @param {Function} callback Callback function to notify the result
+   *  of the operation
+   */
+  function start(host, port, callback) {
```

Comment:
This function is waaaaay too big. It should be splitted in multiple smaller functions. Some advices on how to do it bellow.